File: packages/http-client-csharp/generator/TestProjects/Spector.Tests/Http/Response/StatusCodeRange/StatusCodeRangeTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.ClientModel;
using System.ClientModel.Primitives;
using System.Threading.Tasks;
using NUnit.Framework;
using Response.StatusCodeRange;

namespace TestProjects.Spector.Tests.Http.Response.StatusCodeRange
{
    public class StatusCodeRangeTests : SpectorTestBase
    {
        [SpectorTest]
        public Task StatusCodeNotFound() => Test((host) =>
        {
            var options = new StatusCodeRangeClientOptions();
            var client = new StatusCodeRangeClient(host, options);
            var ex = Assert.ThrowsAsync<ClientResultException>(async() => await client.ErrorResponseStatusCode404Async());
            var error = ModelReaderWriter.Read<NotFoundError>(ex!.GetRawResponse()!.Content);
            Assert.IsNotNull(error);
            Assert.AreEqual("not-found", error!.Code);
            Assert.AreEqual("resource1", error.ResourceId);

            return Task.CompletedTask;
        });

        [SpectorTest]
        public Task StatusCodeInRange() => Test((host) =>
        {
            var options = new StatusCodeRangeClientOptions();
            var client = new StatusCodeRangeClient(host, options);
            var ex = Assert.ThrowsAsync<ClientResultException>(async() => await client.ErrorResponseStatusCodeInRangeAsync());
            var error = ModelReaderWriter.Read<ErrorInRange>(ex!.GetRawResponse()!.Content);
            Assert.IsNotNull(error);
            Assert.AreEqual("request-header-too-large", error!.Code);
            Assert.AreEqual("Request header too large", error.Message);

            return Task.CompletedTask;
        });
    }
}

File: packages/http-client-java/generator/http-client-generator-clientcore-test/src/test/java/response/statuscoderange/StatusCodeRangeTests.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package response.statuscoderange;

import io.clientcore.core.http.models.HttpResponseException;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class StatusCodeRangeTests {

    private final StatusCodeRangeClient client = new StatusCodeRangeClientBuilder().buildClient();

    @Test
    public void testErrorStatusCodeRange() {
        HttpResponseException httpResponseException
            = Assertions.assertThrows(HttpResponseException.class, client::errorResponseStatusCodeInRange);
        Assertions.assertEquals("Request header too large",
            ((ErrorInRange) httpResponseException.getValue()).getMessage());

        httpResponseException
            = Assertions.assertThrows(HttpResponseException.class, client::errorResponseStatusCode404);
        Assertions.assertEquals("resource1", ((NotFoundError) httpResponseException.getValue()).getResourceId());
    }
}
